src/transaction/txParser.ts

```diff
@@ -43,13 +47,51 @@ const parseTxInputs = (
   return txInputs.map(([txHash, outputIndex]): _Input => ({ txHash, outputIndex }))
 }
 
-const parseTxOutputs = (
-  txOutputs: any[],
-): _Output[] => {
+const parseAssets = (assets: any): _Asset[] => {
+  if (!(assets instanceof Map)) {
+    throw Error(Errors.TxAssetParseError)
+  }
+  return Array.from(assets).map(([assetName, coins]) => {
+    if (!Buffer.isBuffer(assetName)) {
+      throw Error()
```

Comment:
Perhaps some particular error? In other places, too.

Reply:
:astonished: Thnx, I forgot as I was refactoring this. Fixed.